### PR TITLE
Allow string aggs for `dask_cudf.CudfDataFrameGroupBy.aggregate`

### PR DESCRIPTION
**Pull request by charlesbluca**
I noticed that `CudfDataFrameGroupBy.aggregate` doesn't actually support passing aggregations as strings, for example something like

```python
import cudf
import dask_cudf

gdf = cudf.DataFrame({'id4': 4*list(range(6)), 'id5': 4*list(reversed(range(6))), 'v3': 6*list(range(4))})
gddf = dask_cudf.from_cudf(gdf, npartitions=5)

gddf.groupby("id4").agg("mean")
```

Would actually end up using the upstream `aggregate` implementation. This is because:

- `CudfDataFrameGroupBy.aggregate` does not convert string aggs to a dict before calling `_is_supported` on them
- `_is_supported` only handles list / dict aggs, returning false otherwise

I've resolved this by adding string support to `_is_supported`, and moving the conversion of aggs to the internal `groupby_agg`.

It looks like this is exposing some failures for `first` and `last` groupby aggs, as tests that were originally using upstream Dask to compute these aggregations (I assume accidentally since these aggregations are listed as supported) are now using dask-cuDF and getting the wrong result.

### CODE REVIEW

File: python/dask_cudf/dask_cudf/groupby.py [hunk 1]
```diff
@@ -423,10 +424,8 @@ def last(self, split_every=None, split_out=1):
     def aggregate(self, arg, split_every=None, split_out=1):
         if arg == "size":
             return self.size()
-        arg = _redirect_aggs(arg)
 
-        if not isinstance(arg, dict):
-            arg = {self._slice: arg}
+        arg = _redirect_aggs(arg)
```
Comment (galipremsagar) [inline python/dask_cudf/dask_cudf/groupby.py:428]: Haven't checked out the code locally but does this already handle the `not dict` case?

Comment (charlesbluca) [reply python/dask_cudf/dask_cudf/groupby.py:428]: Sorry for the delay - was re-familiarizing myself with this branch.

I removed this in an attempt to consolidate the aggregation input normalization entirely within `groupby_agg`, but it looks like using that input normalization would make it so we end up computing the aggregation on all non-groupby columns for series groupbys:

```python
    columns = [c for c in ddf.columns if c not in gb_cols]
    if not isinstance(aggs, dict):
        aggs = {col: aggs for col in columns}
```

Which is unnecessary and inefficient - knowing this, it's best that we add this series groupby specific check back in.

**Review by bdice — APPROVED** [review]
No comments -- looks good to me!

**Review by galipremsagar — APPROVED** [review]

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**